### PR TITLE
Swarm

### PR DESCRIPTION
**Pull request by ehazlett**
This replaces the Citadel / Shipyard integration with Swarm.  It works almost identically using the cluster event stream from Swarm.  This also adds TLS support and the ability to automatically re-connect / reload after a connection drop to Swarm / Docker.

This PR also uses Godeps as recommended - hence the larger changeset.

### CODE REVIEW

**Comment by ehazlett** [comment]
This uses an error channel for the Dockerclient (https://github.com/samalba/dockerclient/pull/68).  The Godeps should be updated once that is merged.

**Comment by vieux** [comment]
@ehazlett time to update the Godeps

**Comment by ehazlett** [comment]
@vieux thx!

File: controller/Godeps/Godeps.json [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
-			"Comment": "beta4-41-g606f8dd",
-			"Rev": "606f8dde3f4dcf416580f496838cabdbbf5786c7"
+			"ImportPath": "github.com/Sirupsen/logrus",
+			"Comment": "v0.6.0-5-gf92b795",
+			"Rev": "f92b7950b372b1db80bd3527e4d40e42555fe6c2"
 		},
 		{
 			"ImportPath": "github.com/samalba/dockerclient",
-			"Rev": "a79250ae5dc96514cf6b06b8a8b7481164da7d4d"
+<<<<<<< Updated upstream
```
Comment (vieux) [inline controller/Godeps/Godeps.json]: @ehazlett I guess it's a merge fail ?

**Comment by vieux** [comment]
Works great!, can you explain a bit more the `INTERLOCK_DATA` variable ?

File: controller/Godeps/Godeps.json (hunk 1 above)
Comment (ehazlett) [inline controller/Godeps/Godeps.json]: oops yeah -- i'm not sure how it compiled with that.  it's fixed now thx!

**Comment by ehazlett** [comment]
@vieux sure.  the `INTERLOCK_DATA` allows for customization of the backend at container runtime.  For example, the following options are available:

* `hostname`: override the container hostname -- this is the combined with the domain to create the endpoint
* `domain`: override the container domain
* `alias_domains`: specify a list of alias domains to add (`{"alias_domains": ["foo.com", "bar.com"]}`)
* `port`: specify which container port to use for backend (`{"port": 8080}`)
* `warm`: connect to the container before adding to the backend (`{"warm": true}`)
* `check`: specify a custom check for the backend (`{"check": "httpchk GET /"}`)
* `backend_options`: specify a list of additional options for the backend (`{"backend_options": ["forceclose", "http-no-delay"]}`) -- see http://cbonte.github.io/haproxy-dconv/configuration-1.5.html#4.1
* `check_interval`: specify the interval (in ms) when to run the health check (`{"check_interval": 10000}`)  default: 5000
* `ssl_only`: configure redirect to SSL for backend (`{"ssl_only": true}`)

You specify one ore more in JSON format in the `INTERLOCK_DATA` environment variable when running the container:

```
docker run -ti \
    -P \
    -d \
    --hostname www.example.com \
    -e INTERLOCK_DATA='{"alias_domains": ["foo.com"], "port": 8080, "warm": true, "ssl_only": true}' \
    ehazlett/go-demo
```

This would run the container `ehazlett/go-demo`.  Interlock would setup the backend www.example.com and and alias foo.com, interlock would use the exposed port that corresponded to the container port 8080, send a request to "warm" the container and setup a redirect to force https.

Make sense?

**Comment by vieux** [comment]
@ehazlett yes thanks

**Comment by vieux** [comment]
I tested and it works great.

**Comment by ehazlett** [comment]
awesome!  thanks for testing!

**Pull request merged**